code/missionui/redalert.cpp

```diff
@@ -936,6 +936,13 @@ void red_alert_bash_ship_status()
 
 				// assign the wave number to the wing
 				wingp->current_wave = rws->latest_wave;
+				
+				// make sure that num_waves can accomodate having extra wings.
+				// this would happen if the mission we are entering has too few waves.
+				if (wingp->num_waves < wingp->current_wave){
+					mprintf(("Red alert warning: make sure that the current mission has at least as many waves as exited last mission. Bashing the number of waves of wing %s to %i!\n", wingp->name, wingp->current_wave));					
```

Comment:
I intended the "make sure..." comment to go above the if(); it is a description of what the code is doing, not a warning to the user.  Anyway your comment is fine too; but "extra wings" is misleading and should be reworded.  Between that and tweaking the `mprintf`, I suggest:

```suggestion
				// make sure that num_waves can accommodate the new wave number.
				// this would happen if the mission we are entering has too few waves.
				if (wingp->num_waves < wingp->current_wave){
					mprintf(("Red alert warning: wing %s in the current mission has fewer waves than last mission. Bashing the number of waves from %i to %i.\n", wingp->name, wingp->num_waves, wingp->current_wave));
```

Reply:
Uh... I'm not sure which if you are referencing.  It is currently over an if.

Sure I misspelled accommodate, but extra wings is not misleading because they are extra wings from the last mission that are not present in the current mission.  You saying "the new wave number" is redundant.  It's like saying, "Make sure the box can accommodate the box".

It's also better to be more direct in the mprintf because the modder is less likely to just overlook the statement since debug logs have so much information.  However it is helpful to add the previous number of waves, so I can do that tomorrow.

Reply:
> Uh... I'm not sure which if you are referencing. It is currently over an if.

In other words, the text that is currently inside the if() (and inside the mprintf) is what I intended to be part of the comment above the if().

> Sure I misspelled accommodate, but extra wings is not misleading because they are extra wings from the last mission that are not present in the current mission. You saying "the new wave number" is redundant. It's like saying, "Make sure the box can accommodate the box".

"Extra wings" could mean something like Delta, Epsilon, etc. in addition to Alpha, Beta, and Gamma.  The term *wings* refers to the named collection of ships.  The term *waves* refers to how many "copies" of the collection there can be.  That's why I said "extra wings" is misleading; the term "wings" is being used in a nonstandard way.

By "the new wave number" I'm referring to the new value of `current_wave`, but I'm open to other phrasing if there's something that would be clearer.  Maybe literally "the new value of current_wave".

> It's also better to be more direct in the mprintf because the modder is less likely to just overlook the statement since debug logs have so much information.

The problem though is that the comment I suggested was intended to describe code, and be viewed by coders.  By putting it in the log, it is viewable by modders, and thus it can easily be interpreted as instructions to modders, when that's not the intention.